### DISCUSSION TITLE
How to turn Jekyll all the way off?

### DISCUSSION THREAD

**Discussion by sidewayss** [community/community#138832] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

How do I turn off Jekyll and have GitHub Pages simply copy files to create the site?

https://github.com/sidewayss/rAF

That's my repo.  It is its own website, with an index.html file in the root folder, and html files in sub-directories as appropriate. I copy it directly from my Apache24/htdocs folder into my local clone of the repo for git and GitHub.  It doesn't need any extra processing.  I also use the underscore character to start the names of some of my javascript files and that messes with Jekyll - just another reason to turn it off.

The other issue is the README.md file in the root folder.  Ideally that would be a symbolic link to the index.html, but I just tried that and Markdown doesn't read straight HTML, at least not mine.  I certainly want GitHub Pages to use the index.html, but I'd also like the README that's displayed in the GitHub page for the repo to display correctly, and it does not now.  Am I going to have to do some kind of HTML to Markdown conversion every time I update the index.html?

**Comment by barishaxxer** [comment]
Create a file named .nojekyll in root directory it will disable jekyll.push your html css js files into a seperate branch called gh-pages.from your repository setting click github pages deploy from a branch and select gh-pages branch.that should solve your problem <3

**Reply by sidewayss** [reply]
Thanks.  A few months back I tried creating various `.nojekyll` files in root and other directories to avoid the "_filename.js" problem, but it never worked.  It's just an empty text file, this `.nojekyll` file?  I'll give it another try later today and post the results here.

What about the README.md linking?  Am I just going to have to convert HTML to Markdown?

**Reply by LiteBrite82** [reply]
The .nojekyll file is just an empty file. There is no file name, its just an extension ".nojekyll", just like ".gitignore"

**Reply by LiteBrite82** [reply]
It should also be in the branch that you run your build from. So if you run your build from `Main` and use automation to deploy to `GH-Pages`, `.nojekyll` needs to be in the `Main` branch before the build command is executed.

**Reply by LiteBrite82** [reply]
Maybe change the `homepage` url on line 22 of package.json

**Reply by sidewayss** [reply]
> Maybe change the homepage url on line 22 of package.json

I'll take a look at that. Pages is currently building from my main branch which has an empty `.nojekyll` file at the root...

**Reply by sidewayss** [reply]
That line was put there automatically by GitHub when I first created the prehistoric version of this repo.  I can see how it might need changing, though if Jekyll is all the way turned off, why is it still looking there? Obviously I do not fully comprehend the Pages process.  If this Pages deployment fails I'll try to switch that to index.html, but right now README.md is a symbolic link to index.html, so it should be OK.

**Reply by LiteBrite82** [reply]
I forked your repo and am testing on my end. Im trying to deploy and am stuck at: `Job is waiting for a hosted runner to come online.`

**Reply by sidewayss** [reply]
> I forked your repo and am testing on my end. Im trying to deploy and am stuck at: `Job is waiting for a hosted runner to come online.`

I have no idea what that means.

**Reply by sidewayss** [reply]
fyi - I am hosting the project on my personal site and it works. I'd still like to get it up and running here.

**Reply by LiteBrite82** [reply]
Awesome; an outage just occurred. https://www.githubstatus.com/history

**Reply by sidewayss** [reply]
> Awesome; an outage just occurred. https://www.githubstatus.com/history

I have a knack for such timing.  Blame me.  I won't mind.

**Reply by sidewayss** [reply]
Or maybe I caused it :-)

**Reply by sidewayss** [reply]
My attempted build just officially failed:
`Internal server error. Correlation ID: 3ee6dcea-4721-4cd7-893f-a7a4b8398d35`

**Reply by LiteBrite82** [reply]
It just came back on line. This is what I get from the fork: 

Deploy: https://litebrite82.github.io/rAFTest/
Repo: https://github.com/LiteBrite82/rAFTest

**Reply by sidewayss** [reply]
That's the right text, but not the right font or formatting.  It's missing linked images too, and the link to the next page goes to an error.  I'm building it myself to see what it does.

Though it looks similar to the README display on my repo's page: https://github.com/sidewayss/rAF
That could maybe be fixed with your suggestion about package.json...

If I open the Dev Tools console there are plenty of errors, all about fonts that I host and that I realize now are not in the repo... I'll have to create a fonts folder for the repo itself, if nothing more.

**Reply by sidewayss** [reply]
And I will need to change all the paths everywhere fonts are loaded, a CSS file or two and a .js file.  I am currently using absolute paths referring to my personal root htdocs folder, one level above this project.  Is there a way to use absolute paths, starting at the root, e.g. "/whatever/font.woff2", or should I be using relative paths for everything?

**Reply by LiteBrite82** [reply]
index.html: line 11: `<object data="/raf/img/rAFoffset.svg"`
Thats a local path

**Reply by LiteBrite82** [reply]
Im not familiar with Apache, but it looks like you may need to edit the src references from their previous local instances to the new web based instances.

**Reply by sidewayss** [reply]
> index.html: line 11: <object data="/raf/img/rAFoffset.svg"
Thats a local path

Except that I have no access to anything above `"/raf/"`, to which I'm adapting by adding `"/raf/fonts/"`.  Ok, this means less change.  My current setup bears some resemblance to the Pages result. I'll just be changing `"/fonts/name.ttf"` `"/raf/fonts/name.ttf"`.

But that SVG image isn't showing up either, so maybe relative paths are the better option...

> Im not familiar with Apache, but it looks like you may need to edit the src references from their previous local instances to the new web based instances.

Apache has a root folder for the site named `htdocs`, on my hosting service it's `public_html`.  I deploy this project as a sub-folder of the root web page, in `htdocs/raf`.  I have fonts in a `htdocs/fonts` folder.  I think that relative paths, like `fonts/name.ttf`, or more explicitly `./fonts/name.ttf`, should work around the issue without any further side-effects.  I'll post here when I've given it a go.

Thanks.

**Reply by sidewayss** [reply]
@LiteBrite82 - Yes, that is much better.  The homepage for the project comes up nicely now:
https://sidewayss.github.io/rAF/
But farther down the tree I have another issue with a common set of files, common for my whole site right now, some custom HTML elements.  I'll have to move or copy that folder into `raf/html-elements` just like `raf/fonts`  

And there are some images in a downstream page that need their `src` attribute adjusted.  I'm getting there...

**Reply by sidewayss** [reply]
What's unfortunate about this is that I already have a separate github repo for `html-elements` here:
https://github.com/sidewayss/html-elements

I'll be adding them as a sub-directory in this repo just to get them to work with Pages.  Hmmm.  Any way around that?  I'm not anticipating a positive answer to that one.

I'm going to have to either copy the files into a duplicate directory, and manually update them in order to update this Pages site if there are changes.  Or I'm going to have to import these modules from GitHub itself, or my personal site - change some HTML and JavaScript code.  Either way, I'll pick one.  I think the former is better because it isolates the issue to the Pages deployment of the project's sub-site.  I'll sleep on it...

Thanks again for your help.

**Reply by LiteBrite82** [reply]
yeah, you would have to add them to the repo, or find an online reference to them.
https://developers.google.com/fonts/docs/getting_started

**Comment by sidewayss** [comment, marked as answer]
After sleeping on it, I'm changing my mind.  I don't like adding duplicate files/folders to this repo just to get Pages working properly. I'm going to have to either abandon Pages while I continue to host it myself, or source my fonts and `html-elements` from a common, absolute path source that starts with "https://".  I'll figure it out later today.

To summarize and answer the question:
- You can see my result here: https://sidewayss.github.io/rAF/
- Adding an empty text file named `.nojekyll` to the root folder of my repo did work correctly and bypassed  Jekyll.
- I also had two additional, structural issues with compatibility between a standalone hosting of my repo's sub-site and a multi-user service like Pages. I have two directories in the root directory of my web site:
a) Custom HTML elements . I already have an `html-elements` repo, so I deployed it to Pages and that solved it.
b) Fonts.  I initially created a new `fonts` repo and deployed it to Pages, but then I discovered [this gist](https://gist.github.com/codeadamca/bc3dcb6fc768ee1714918aa15cf13fad) that describes how to take control the root folder of your Pages deployment.
So I created a new repo: `sidewayss.github.io`. No need to configure Pages, it's automatic with that name format. In my initial commit the repo contains a fonts folder and a favicon.ico.  It's the 404 error for the favicon (in the Dev Tools console) that led me to that gist.

The &lt;USERNAME&gt;.github.io repo is the correct solution for a situation like mine, that is already setup as a web site, or for anyone who wants to configure their favicon :)  You have one core repo that *is* your Pages hompage's root directory.  For example, I was able to revert to my original `@font-face {src:"path"}` path format:
```
@font-face {
    font-family:"Averia";
    src:url("/fonts/AveriaSerifLibre-Regular.ttf") format("truetype");
}
```
At some point I'll delete my old `homepage` repo and move it inside this new one.

**Comment by ngchthien** [comment]
You're right, Jekyll can sometimes get in the way when you just want simple static file hosting! Here's how to turn Jekyll completely off and address your README.md situation:
1. Create an Empty .nojekyll File
In the root directory of your GitHub Pages repository (sidewayss/rAF), create a new file named .nojekyll. You don't need to put anything inside it; an empty file is enough.
Commit this file and push it to your GitHub repository.
This tells GitHub Pages to bypass Jekyll processing entirely and treat your repository as a plain website.
2. Handling README.md and index.html
You have a couple of options here:
Option A: Separate Files (Simpler)
Keep your index.html for your website.
Create a dedicated README.md with content specifically formatted for Markdown display on your GitHub repo page. This gives you more control over how each file looks in its respective context.
Option B: JavaScript Redirect (More Technical, Less Ideal)
Keep only your index.html.
In your index.html, add a JavaScript redirect at the very top of the <head> section:
<script>
  if (window.location.pathname === "/rAF/") { // Adjust path if needed
    window.location.replace("/rAF/index.html"); // Adjust path if needed
  }
</script>
Use code [with caution](https://support.google.com/legal/answer/13505487).
Html
This attempts to redirect users from the raw README.md view to your actual index.html. However, this can be unreliable and might not be ideal for SEO.
3. Push Your Changes
Make sure to commit and push your changes to GitHub. Your site should now be served as a pure static site without any Jekyll interference.
Important Notes:
Underscores: With Jekyll disabled, your JavaScript files starting with underscores (_) will no longer cause issues.
Caching: Sometimes GitHub Pages can take a little while to reflect changes after disabling Jekyll. If you don't see the update right away, wait a few minutes or clear your browser cache.
I hope this helps! Let me know if you have any more questions.

**Comment by sidewayss** [comment]
@ThienNg0 - thanks for the redirect suggestion.  Very interesting, but given the unreliability I think I'll pass.  I'm going to setup a separate README.md written specifically for the github repo page, with links to other pages on Pages or my personal site, as you recommend.

**Comment by sidewayss** [comment]
@LiteBrite82 fyi - I updated my answer above because I came up with a proper solution: create a "fonts" repo and deploy it to Pages along with my `html-elements` repo.  Voilà!

**Comment by sidewayss** [comment]
OK, sorry for the churn, but I found a more correct solution for the fonts in this gist: https://gist.github.com/codeadamca/bc3dcb6fc768ee1714918aa15cf13fad

I implemented it successfully and updated my answer above to provide new detail.  Note the favicon too.
